modules/get_genes.nf

```diff
@@ -13,43 +13,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/** 2) Add the clustering stage to the workflow
- * 
- * This second stage of the workflow aims to divide the sequences into genes 
- * (some genes are highly conserved within the same genus and can be used to find those sequences of dubious quality).
- *
-**/
-
-nextflow.enable.dsl = 2
-
 process GET_GENES {
-   
-   publishDir "./data/${name}/cluster", mode: 'copy', overwrite: false
-   
-   input:
-   path genes
-   val name
- 
-   output:
-   path '*.fasta'
-   
-   shell:
-   """
-   get_genes -i "$genes" -o "$name"
-   """
-}
+    tag "$file_path"
+    publishDir "${params.output_dir}/cluster", mode: 'copy', overwrite: false
+
+    input:
+        tuple path(file_path), val(file_format)
 
-workflow {
-   if (params.name == null) {
-      print("Please, insert the output file name to cluster")
-      exit(1)
-   }
+    output:
+        tuple path('*.fasta'), val('fasta')
```

Comment:
That's correct. At this moment this function is quite limited in its configuration and modules used: https://github.com/JAlvarezJarreta/MEvoLib/blob/main/mevolib/cluster/__init__.py#L112-L120. It will need to be expanded in the future.